File: Softwaresystems/Python AlgoDat/Algorithms/Searching/binary_search.py
def binary_search(arr, low, high, x):

    if high >= low:

        mid = (high + low) // 2

        if arr[mid] == x:
            return mid

        elif arr[mid] > x:
            return binary_search(arr, low, mid - 1, x)

        else:
            return binary_search(arr, mid + 1, high, x)

    else:
        return -1

# Test array
arr = [12,45,78,98,63,21,458,78,96,420,69,21,78,96,55,266,666,10,214,789,654,1,2,4,8,9,3,5,78]
x = 10

# Function call
result = binary_search(arr, 0, len(arr)-1, x)

if result != -1:
    print("Element is present at index", str(result))
else:
    print("Element is not present in array")

File: Softwaresystems/Informatik2/pre-Semester Practice/Algorithms/Search/binary_search.c
#include <string.h>
#include <stdio.h>

int binarySearch(int arr[], int l, int r, int n){
    
    if(r >= l) {
        int mid = l + (r-1)/2;

        if(arr[mid] == n) {
            return mid;
        }

        if (arr[mid] > n) {
            return binarySearch(arr, l, mid - 1, n);
        }

        return binarySearch(arr, mid, r-1, n);
    }

    return -1;
}

int main(void) {
    int arr[] = {5, 210, 124, 122, 5, 89, 1, 2, 19, 69, 420, 31, 2001, 12, 31, 41, 690, 128};
    int n = 31; 
    int r = sizeof(arr)/sizeof(arr[0]); 

    int result = binarySearch(arr, 0, r - 1, n);

    if(result == -1){
        printf("Element is not present in the array");
    } 
    else {
        printf("Element is present at index %d", result);
    }
    return 0;
}